trinity/plugins/builtin/ethstats/ethstats_client.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import datetime
+import json
+import platform
+
+import websockets
+
+from trinity.utils.version import construct_trinity_client_identifier
+
+class EthstatsClient:
+    def __init__(self, websocket: websockets.client.WebSocketClientProtocol, node_id: str, server_url: str, server_secret: str) -> None:
+        self.websocket = websocket
+        self.send_queue = asyncio.Queue()
+
+        self.node_id = node_id
+        self.server_url = server_url
+        self.server_secret = server_secret
+
+        self.client = construct_trinity_client_identifier()
+        self.os = platform.system()
+        self.os_v = platform.release()
+
+    async def response_handler(self) -> None:
+        async for message in self.websocket:
+            command, data = self.stat_recv(message)
+            if command == 'node-pong':
+                timestamp = round(datetime.datetime.now().timestamp() * 1000)
+                latency = (timestamp - data['clientTime']) // 2
+                await self.send_latency(latency)
+    
+    async def request_handler(self) -> None:
+        while True:
+            message = await self.send_queue.get()
+            [command, data] = message
+            await self.stat_send(command, data)
+
+    async def connection_handler(self) -> None:
```

Comment:
It might be good to move this into the `EthStatsService` so that you can use `BaseService.run
_task()` to run the `request_handler` and `response_handler`.  Otherwise, they won't be properly shutdown.

Reply:
Yes, either that or derive the `EthstatsClient` from `BaseService` as well and pass the token from the `EthStatsService` so that it will shutdown in tandem.

Reply:
@cburgdorf That's approach I've working on now, it seems a bit better to handle responses/requests in client completely and EthstatsService will handle actually getting stats and sending them to client.

Reply:
Reimplemented `EthstatsClient` as child service.